packages/contracts/contracts/optimistic-ethereum/ovm/SafetyChecker.sol

```diff
@@ -63,136 +35,102 @@ contract SafetyChecker is ContractResolver {
-                        // 1. PUSH1 value
-                        // 2. PUSH20 execution manager address
-                        // 3. PUSH or DUP gas
-                        // 4. CALL
+            // inline assembly removes the extra add + bounds check
+            assembly {
+                let tmp := mload(_pc)
 
-                        // if opIndex < 3, there would be 0s here, so we don't need the check
-                        uint256 gasOp = (ops >> 8) & 0xFF;
-                        uint256 addressOp = (ops >> 16) & 0xFF;
-                        uint256 valueOp = (ops >> 24) & 0xFF;
-                        if (
-                            gasOp < 0x60 || // PUSHes are 0x60...0x7f
-                            gasOp > 0x8f || // DUPs are 0x80...0x8f
-                            addressOp != 0x73 || // address must be set with a PUSH20
-                            valueOp != 0x60 // value must be set with a PUSH1
-                        ) {
-                            return false;
-                        } else {
-                            uint256 pushedBytes;
-                            // gas is set with a PUSH##
-                            if (gasOp >= 0x60 && gasOp <= 0x7f) {
-                                pushedBytes = gasOp - 0x5f;
-                            }
+                let mpc := byte(0, mload(add(skip, byte(0, tmp))))
+                mpc := add(mpc, byte(0, mload(add(skip, byte(mpc, tmp)))))
+                mpc := add(mpc, byte(0, mload(add(skip, byte(mpc, tmp)))))
+                mpc := add(mpc, byte(0, mload(add(skip, byte(mpc, tmp)))))
+                mpc := add(mpc, byte(0, mload(add(skip, byte(mpc, tmp)))))
+                mpc := add(mpc, byte(0, mload(add(skip, byte(mpc, tmp)))))
+                _pc := add(_pc, mpc)
 
-                            // 23 is from 1 + PUSH20 + 20 bytes of address + PUSH or DUP gas
-                            byte callValue = _bytecode[pc - (23 + pushedBytes)];
-
-                            // 21 is from 1 + 19 bytes of address + PUSH or DUP gas
-                            address callAddress = toAddress(_bytecode, (pc - (21 + pushedBytes)));
+                op := byte(mpc, tmp)
+            }
 
-                            // CALL is made to the execution manager with msg.value of 0 ETH
-                            if (callAddress != address(resolveExecutionManager()) || callValue != 0 ) {
-                                return false;
-                            }
+            // + push opcodes
+            // + stop opcodes [STOP(0x00),JUMP(0x56),RETURN(0xf3),REVERT(0xfd),INVALID(0xfe)]
+            // + caller opcode CALLER(0x33)
+            // + blacklisted opcodes
+            uint256 opBit = 1 << op;
+            if (opBit & _opcodeProcMask == 0) {
+                if (opBit & _opcodePushMask == 0) {
+                    // subsequent bytes are not opcodes. Skip them.
+                    _pc += (op - 0x5e);
```

Comment:
might create a const for `0x5e`. I assume that's the opcode before `PUSH1`? Maybe named `pushStart`?